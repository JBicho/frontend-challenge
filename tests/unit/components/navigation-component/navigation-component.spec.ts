import NavigationComponent from '@/components/Navigation-component/Navigation-component.vue';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuex);

describe('Add Row Component Tests', () => {
  let wrapper: any;
  let store: any;
  let actions: any;
  let state: any;

  beforeEach(() => {
    actions = {
      getTranscriptions: jest.fn(() => true),
      uploadTranscriptions: jest.fn(() => true)
    };

    state = {
      transcriptionList: [],
      errors: []
    };

    store = new Vuex.Store({
      state: {},
      actions,
      modules: {
        transcription: {
          namespaced: true,
          state,
          actions
        }
      }
    });
  });

  test('Snapshot test', () => {
    const Constructor = Vue.extend(NavigationComponent);
    const vm = new Constructor().$mount();
    expect(vm.$el).toMatchSnapshot();
  });

  test('Should pass action to the store on button click', () => {
    wrapper = shallowMount(NavigationComponent, {
      store,
      localVue
    });

    wrapper.find('.upload-transcriptions').trigger('click');

    expect(actions.uploadTranscriptions).toHaveBeenCalled();

    wrapper.find('.get-transcriptions').trigger('click');

    expect(actions.getTranscriptions).toHaveBeenCalled();
  });
});
