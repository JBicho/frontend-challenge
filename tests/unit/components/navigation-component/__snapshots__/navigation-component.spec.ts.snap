// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Row Component Tests Snapshot test 1`] = `
<nav
  class="navigation"
>
  <div
    class="navigation--wrapper"
  >
    <h2
      class="navigation--app-title"
    >
      Transcription
    </h2>
     
    <ul
      class="navigation--buttons"
    >
      <li>
        <button
          class="upload-transcriptions navigation--buttons--button"
        >
          <svg
            class="svg-icon svg-fill"
            style="width: 1.5rem; height: 1.5rem;"
            version="1.1"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                pid="0"
              />
              <path
                d="M9 16h6v-6h4l-7-7-7 7h4v6zm-4 2h14v2H5v-2z"
                fill="#8F97A8"
                fill-rule="nonzero"
                pid="1"
              />
            </g>
          </svg>
        </button>
      </li>
       
      <li>
        <button
          class="get-transcriptions"
        >
          <svg
            class="svg-icon svg-fill"
            style="width: 1.5rem; height: 1.5rem;"
            version="1.1"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                pid="0"
              />
              <path
                d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9h-4v4h-2v-4H9V9h4V5h2v4h4v2z"
                fill="#8F97A8"
                fill-rule="nonzero"
                pid="1"
              />
            </g>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
